compiler/parser/RdrHsSyn.hs

```diff
@@ -2328,6 +2328,32 @@ asPatPrefixCon varName conName =
     (noLoc $ mkRdrUnqual (mkVarOcc varName))
     (noLoc $ XPat (noLoc $ ConPatIn conName $ PrefixCon [])))
 
+-- | The "-Wunused-matches hack." Construct dummy bindings of the form
+-- '_ = this@T{..}' or '_ = self'. We use this hack to suppress
```

Comment:
What is the `@T{..}` about? That looks more like a pattern than the RHS of a `let`-binding to me.
```suggestion
-- @let _ = this in ...@ or @let _ = self in ...@. We use this hack to suppress
```

Reply:
That is the example shape of the incoming 'as pattern' "this" argument for which a dummy bind of the name this is created.

Reply:
You are right. I see how the comment is confusing. The `=` sign shouldn't be a code literal.

Reply:
I shouldn't try to respond when half asleep. You are right. The suggested fix is right.

Reply:
PR https://github.com/digital-asset/ghc/pull/52.